### PR TITLE
New base64encode and sha256 filters

### PR DESCRIPTION
**Pull request by silviuvergoti**
Adding new filters.
Fixes issues  #573 and #572.

### CODE REVIEW

**Comment by objectProgrammer** [comment]
Perhaps base64 is common enough to be in the core, but if SHA256 gets added then suddenly we could end up with MD5, SHA-1, SHA-2, SHA-3 and more. I feel like there are too many hashing functions for any of them to be included in the core.

Just curious, what's the use for having a hashing filter?

And base64encode sounds too long, maybe just `base64`?

**Comment by silviuvergoti** [comment]
Sure, I'll shorten it. I thought that a simple `base64` would not be clear enough if it concerns encoding or decoding.  
Regarding the hashing filters, there are a lot of web services requiring hashing of PII (emails, ids, phones). If someone needs it, the filter could be changed to something like `{{ originalstring | hash('SHA256') }}` and it could support many more hash algorithms.

File: pebble/src/main/java/com/mitchellbosecke/pebble/extension/core/CoreExtension.java [hunk 1]
```diff
@@ -149,6 +149,8 @@
     filters.put(ReplaceFilter.FILTER_NAME, new ReplaceFilter());
     filters.put(MergeFilter.FILTER_NAME, new MergeFilter());
     filters.put(SplitFilter.FILTER_NAME, new SplitFilter());
+    filters.put(Base64EncoderFilter.FILTER_NAME, new Base64EncoderFilter());
```
Comment (ebussieres) [inline pebble/src/main/java/com/mitchellbosecke/pebble/extension/core/CoreExtension.java:152]: Should we provide a base64 decoder too ?

File: pebble/src/test/java/com/mitchellbosecke/pebble/CoreFiltersTest.java [hunk 2]
```diff
@@ -1329,6 +1329,57 @@ void testReplaceFilterInTemplate() throws PebbleException, IOException {
     assertEquals("I like foo and bar.", writer.toString());
   }
 
+  /**
+   * Tests {@link com.mitchellbosecke.pebble.extension.core.Base64EncoderFilter} if the base64 encoding filter is working for a string value, a string constant, null.
+   */
+  @Test
+  void testBase64EncoderFilterInTemplate() throws PebbleException, IOException {
```
Comment (ebussieres) [inline pebble/src/test/java/com/mitchellbosecke/pebble/CoreFiltersTest.java:1336]: Add a unit test with null as value passed to base64 filter

File: pebble/src/main/java/com/mitchellbosecke/pebble/extension/core/CoreExtension.java (hunk 1 above)
Comment (silviuvergoti) [reply pebble/src/main/java/com/mitchellbosecke/pebble/extension/core/CoreExtension.java:152]: @ebussieres, decoding a base64 string will return a byte array. If it's a binary object, reconstructing it might not be feasible. Do you suggest a return value type of `byte[]`? The encoder was mainly targeted to strings with special characters.

Comment (ebussieres) [reply pebble/src/main/java/com/mitchellbosecke/pebble/extension/core/CoreExtension.java:152]:
```
byte[] decodedBytes = Base64.getDecoder().decode(encodedString);
String decodedString = new String(decodedBytes);
```

File: pebble/src/main/java/com/mitchellbosecke/pebble/extension/core/Base64EncoderFilter.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+import java.util.Base64;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This class implements the 'base64encode' filter.
+ *
+ * @author Silviu Vergoti
+ */
+public class Base64EncoderFilter implements Filter {
```
Comment (ebussieres) [inline pebble/src/main/java/com/mitchellbosecke/pebble/extension/core/Base64EncoderFilter.java:18]: Can you add some documentation for the new filters too ? 

https://github.com/PebbleTemplates/pebble/tree/master/docs/src/orchid/resources

File: pebble/src/main/java/com/mitchellbosecke/pebble/extension/core/CoreExtension.java (hunk 1 above)
Comment (objectProgrammer) [reply pebble/src/main/java/com/mitchellbosecke/pebble/extension/core/CoreExtension.java:152]: What's the use case? I mean base64 encoding can be useful for putting in a link or showing a raw image but decoding seems pretty useless, if it's data from another server I'd expect it to be decoded before being provided to the template. Besides, new String(...) is loose and will probably cause problems on Windows machines, StandardCharsets.UTF_8 should be supplied.

Comment (silviuvergoti) [reply pebble/src/main/java/com/mitchellbosecke/pebble/extension/core/CoreExtension.java:152]: I'll add the decoder and it will try to return an UTF-8 String. If there's an exception, we'll throw it. No object casting will be done.

File: pebble/src/main/java/com/mitchellbosecke/pebble/extension/core/Base64EncoderFilter.java (hunk 3 above)
Comment (silviuvergoti) [reply pebble/src/main/java/com/mitchellbosecke/pebble/extension/core/Base64EncoderFilter.java:18]: Sure.
